models/models-games.js

```diff
@@ -1,54 +1,93 @@
-const { query } = require('../db/connection');
-const db = require('../db/connection');
+const { query } = require("../db/connection");
+const db = require("../db/connection");
 
 exports.selectCategories = () => {
     return db.query(`SELECT * FROM categories`).then((categories) => {
         return categories.rows;
-    })
+    });
 };
 
 exports.selectReviewByID = (review_id) => {
-    return db.query(`SELECT reviews.*, CAST (COUNT(comments) AS INTEGER) AS comment_count
+    return db
+        .query(
+            `SELECT reviews.*, CAST (COUNT(comments) AS INTEGER) AS comment_count
     FROM reviews
     LEFT JOIN comments
     ON reviews.review_id = comments.review_id
     WHERE reviews.review_id = $1
-    GROUP BY reviews.review_id`, [review_id]).then((review) => {
-        return review.rows[0]
-    })
+    GROUP BY reviews.review_id`,
+            [review_id]
+        )
+        .then((review) => {
+            return review.rows[0];
+        });
 };
 
 exports.updateReviewByID = (review_id, votes) => {
-    return db.query('UPDATE reviews SET votes = votes + $1 WHERE review_id = $2 RETURNING *', [votes, review_id]).then((review) => {
-        return review.rows[0]
-
-    })
-}
+    return db
+        .query(
+            "UPDATE reviews SET votes = votes + $1 WHERE review_id = $2 RETURNING *",
+            [votes, review_id]
+        )
+        .then((review) => {
+            return review.rows[0];
+        });
+};
 
 exports.selectUsers = () => {
     return db.query(`SELECT * FROM users`).then((users) => {
         return users.rows;
-    })
+    });
 };
 
-exports.selectReviews = () => {
-    return db.query(`SELECT reviews.owner, reviews.review_id, reviews.created_at, reviews.title, reviews.category, reviews.review_img_url, reviews.votes, CAST (COUNT(comments) AS INTEGER) AS comment_count
+exports.selectReviews = (sort_by = "created_at", order = "desc", category) => {
+    let queryValues = [];
+    const validSortBy = ["created_at", "owner", "category", "review_id"];
+    let queryString = `SELECT reviews.owner, reviews.review_id, reviews.created_at, reviews.title, reviews.category, reviews.review_img_url, reviews.votes, CAST (COUNT(comments) AS INTEGER) AS comment_count
     FROM reviews
     LEFT JOIN comments
-    ON reviews.review_id = comments.review_id
-    GROUP BY reviews.review_id
-    ORDER BY created_at DESC`).then((reviews) => {
+    ON reviews.review_id = comments.review_id`
+    if (category) {
+        queryValues.push(category)
+        queryString += ` WHERE category = $${queryValues.length}`;
+    }
+    queryString += ` GROUP BY reviews.review_id`;
+    if (validSortBy.includes(sort_by)) {
+        if ((order === "asc")) {
+            queryString += ` ORDER BY ${sort_by} ASC`;
```

Comment:
Try to avoid using string templates in this way as it leaves you open to SQL injection :)

Reply:
Hello! Usually, I would definitely agree that this is unsafe, but here ASC and DESC are hard-coded and I am unsure why this would leave us open to SQL injections, when we already checked sort_by, and it can only be one of the 4 things in our validSortBy Array.
I am definitely happy to parameterise it, but I did have a long thought about doing it like this which is why I decided to go ahead with it.
Happy for any input as to why that might not be the correct way!

Reply:
Apologies you are completely correct! My brain is a bit mushy after this week.

Ignore my comment this way is completely fine :)